packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1673,6 +1676,60 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this.context.getAbsoluteUrl(relativeUrl);
     }
 
+    /**
+     * Find and reset nodes whose unreferenced state, such as unreferenced time, needs to be reset. These are nodes
+     * that transitioned from `unreferenced -> referenced -> unreferenced` since the last GC run.
+     * @param resultFromCurrentRun - The result of the current GC run.
+     */
+    private resetUnreferencedState(resultFromCurrentRun: IGCResult) {
```

Comment:
Any chance we can group all GC-related methods, variables and refactor it into its own class, such that ContainerRuntime only need to know about few entry points, like collectGarbage()?

Reply:
I was actually doing something similar but in summarizer node - Add a RootSummarizerNode class which handles all GC related functionality of the container runtime. That is turning out to be more complicated than what I want. So maybe creating a class off of container runtime might be simpler.

Reply:
Created an issue for this - https://github.com/microsoft/FluidFramework/issues/8126. I will be working on this before I merge this PR.